src/helpers/findAll.ts

```diff
@@ -0,0 +1,39 @@
+import { ReactTestInstance } from 'react-test-renderer';
+import { AccessibilityOption } from '../queries/accessibilityOption';
+import { isInaccessible, isSubtreeInaccessible } from './accessiblity';
+
+export function findAll<Options extends AccessibilityOption>(
+  instance: ReactTestInstance,
+  predicate: (node: ReactTestInstance) => boolean,
+  options?: Options
+) {
+  const results = instance.findAll(predicate);
+
+  //TODO: implement getConfig
+  const defaultHidden = true;
+  const hidden = options?.hidden ?? defaultHidden; // We will want to add `defaultHidden: boolean` option to `configure`
+
+  if (hidden) {
+    return results;
+  }
+
+  const subtreeIsInaccessibleCache = new WeakMap<ReactTestInstance>();
+  function cachedIsSubtreeInaccessible(element: ReactTestInstance | null) {
+    if (element === null) {
+      return true;
+    }
+
+    if (!subtreeIsInaccessibleCache.has(element)) {
+      subtreeIsInaccessibleCache.set(element, isSubtreeInaccessible(element));
+    }
+
+    return subtreeIsInaccessibleCache.get(element);
+  }
+
+  return results.filter(
+    (result) =>
+      !isInaccessible(result, {
+        isSubtreeInaccessible: cachedIsSubtreeInaccessible,
+      })
+  );
```

Comment:
I wonder if it wouldn't be a better API if instead of `isInaccessible` didnt take `isSubtreeInaccessible` but rather it would take a cache option `cache?: WeakMap<ReactTestInstance>`, and it would just pass that cache to `isSubtreeInaccessible` and we would also put the cache hit logic there. Wdyt @MattAgn?

Reply:
That would deviate from DTL's option but that option feels kinda internal anyway

Reply:
totally agree!

Reply:
will do that